rllib/models/torch/torch_distributions.py

```diff
@@ -264,13 +272,266 @@ def kl(self, other: "Distribution", **kwargs) -> TensorType:
+    def sample(self) -> TensorType:
+        arr = [cat.sample() for cat in self.cats]
+        sample_ = torch.stack(arr, dim=1)
+        return sample_
+
+    @override(Distribution)
+    def rsample(self, sample_shape=()):
+        # TODO (Kourosh) Implement Categorical sampling using grad-passthrough trick.
+        raise NotImplementedError
+
+    @override(Distribution)
+    def logp(self, value: torch.Tensor) -> TensorType:
+        value = torch.unbind(value, dim=1)
+        logps = torch.stack([cat.log_prob(act) for cat, act in zip(self.cats, value)])
+        return torch.sum(logps, dim=0)
+
+    @override(Distribution)
+    def entropy(self) -> TensorType:
+        return torch.sum(
+            torch.stack([cat.entropy() for cat in self.cats], dim=1), dim=1
+        )
+
+    @override(Distribution)
+    def kl(self, other: Distribution) -> TensorType:
+        kls = torch.stack(
+            [
+                torch.distributions.kl.kl_divergence(cat, oth_cat)
+                for cat, oth_cat in zip(self.cats, other.cats)
+            ],
+            dim=1,
+        )
+        return torch.sum(kls, dim=1)
+
+    @staticmethod
+    @override(Distribution)
+    def required_input_dim(space: gym.Space, **kwargs) -> int:
+        return int(np.sum(space.nvec))
+
+    @classmethod
+    @override(Distribution)
+    def from_logits(
+        cls,
+        logits: torch.Tensor,
+        input_lens: List[int],
+        temperatures: List[float] = None,
+        **kwargs,
+    ) -> "TorchMultiCategorical":
+        """Creates this Distribution from logits (and additional arguments).
+
+        If you wish to create this distribution from logits only, please refer to
+        `Distribution.get_partial_dist_cls()`.
+
+        Args:
+            logits: The tensor containing logits to be separated by logit_lens.
+                child_distribution_cls_struct: A struct of Distribution classes that can
+                be instantiated from the given logits.
+            input_lens: A list of integers that indicate the length of the logits
+                vectors to be passed into each child distribution.
+            temperatures: A list of floats representing the temperature to use for
+                each Categorical distribution. If not provided, 1.0 is used for all.
+            **kwargs: Forward compatibility kwargs.
+        """
+        if not temperatures:
+            # If temperatures are not provided, use 1.0 for all actions.
+            temperatures = [1.0] * len(input_lens)
+
+        categoricals = [
+            TorchCategorical(logits=logits, temperature=temperature)
+            for logits, temperature in zip(
+                torch.split(logits, input_lens, dim=1), temperatures
+            )
+        ]
+
+        return TorchMultiCategorical(categoricals=categoricals)
+
+    def to_deterministic(self) -> "TorchMultiDistribution":
+        return TorchMultiDistribution([cat.to_deterministic() for cat in self.cats])
```

Comment:
Note that this PR does not test to_deterministic for most distributions. Lmk if I should add such tests.